Training24/Test.c

```diff
@@ -1,19 +1,172 @@
-// ------------------------------------------------------------------------------------
+// ---------------------------------------------------------------------------------
 // Training ~ A training program for new joiners at Metamation, Batch - July 2024.
 // Copyright (c) Metamation India.
-// -------------------------------------
-//  Dhaarani Mohanasundaram
-// -------------------------------------
-// Standard Test file in Main
-// -------------------------------------
+// Dhaarani Mohanasundaram
+// ----------------------------------------------------------------
 // Test.c
-// Program on main branch.
-// ------------------------------------------------------------------------------------
+// Program on branch A4.
+// Program that implements tests for the phrase or integer for a palindrome.
+// ---------------------------------------------------------------------------------
 #include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdbool.h>
 #include "Header.h"
 
-int main() {
-const char* message = GetMessage ();
-printf ("%s\n", message);
-return;
-}
\ No newline at end of file
+#define RED "\x1b[31m"
+#define GREEN "\x1b[32m"
+#define YELLOW "\x1b[33m"
+#define RESET "\x1b[0m"
+
+/// <summary>Function to print the table header for Phrase Testcases.</summary>
+void PhraseTableHeader ();
+
+/// <summary>Function to print a single row in the table for phrase.</summary>
+void PhraseTableRow (const char* input, const char* result);
+
+/// <summary>Function to print the table header for Integer Testcases.</summary>
+void IntegerTableHeader ();
+
+/// <summary>Function to print a single row in the table for integer.</summary>
+void IntegerTableRow (int number, const char* outputStr, bool isPalindrome);
+
+/// <summary>Function to test the user input for phrase palindrome.</summary>
+void TestUserPhrase ();
+
+/// <summary>Function to test the user input for reverse number palindrome.</summary>
+void TestUserInteger ();
+
+/// <summary>Function to clear the new line input characters.</summary>
+void clearInputBuffer ();
+
+int main () {
+   TestPhrasePalindromes ();
+   TestReverseNumbers ();
+   printf ("\n1. In Phrase, empty strings are considered as palindrome.\n2. In Integers, negative integers are not considered as a palindrome.\n");
+
+   while (1) {
+      int choice;
+      char option[100];
+      printf ("\nChoose an option:\n1. Test Phrase\n2. Test Integer\n3. Exit\nEnter your choice (1 or 2 or 3): ");
+      if (!fgets (option, sizeof (option), stdin) || option[0] == '\n' || sscanf (option, "%d", &choice) != 1 || choice < 1 || choice > 3) {
+         printf ("Invalid choice. Please enter 1, 2, or 3.\n");
+         continue;
+      }
+      switch (choice) {
+      case 1:
+         TestUserPhrase ();
+         break;
+      case 2:
+         TestUserInteger ();
+         break;
+      case 3:
+         printf ("Exiting program.\n");
+         return 0;
+      }
+   }
+   return 0;
+}
+
+void PhraseTableHeader () {
```

Comment:
Function name must convey the purpose of the function. It must be PrintTableHeader (). Make this change wherever applicable.
Also, for printing headers one function is enough. No need of separate functions for printing integer headers and phrase headers.